shanoir-ng-studies/src/main/java/org/shanoir/ng/configuration/amqp/RabbitMQSubjectService.java

```diff
@@ -165,6 +183,42 @@ public Long createSubject(String subjectAsString) throws JsonProcessingException
 		}
 	}
 
+	/**
+	 * Receives a shanoirEvent as a json object, concerning a preclinical subject deletion
+	 * @param eventAsString the task as a json string.
+	 */
+	@RabbitListener(bindings = @QueueBinding(
+			key = ShanoirEventType.DELETE_PRECLINICAL_SUBJECT_EVENT,
+			value = @Queue(value = RabbitMQConfiguration.DELETE_ANIMAL_SUBJECT_QUEUE, durable = "true"),
+			exchange = @Exchange(value = RabbitMQConfiguration.EVENTS_EXCHANGE, ignoreDeclarationExceptions = "true",
+					autoDelete = "false", durable = "true", type= ExchangeTypes.TOPIC))
+	)
+
+	@Transactional
+	public void deletePreclinicalSubject(String eventAsString) throws AmqpRejectAndDontRequeueException {
+		SecurityContextUtil.initAuthenticationContext("ADMIN_ROLE");
+		try {
+
+			ShanoirEvent event = mapper.readValue(eventAsString, ShanoirEvent.class);
+
+			Optional<Subject> subject = subjectRepository.findById(Long.valueOf(event.getObjectId()));
+
+			if(subject.isEmpty()){
+				return;
+			}
+
+			String id = subject.get().getId().toString();
+			subjectRepository.delete(subject.get());
+			eventService.publishEvent(new ShanoirEvent(ShanoirEventType.DELETE_SUBJECT_EVENT, id, KeycloakUtil.getTokenUserId(), "", ShanoirEvent.SUCCESS));
```

Comment:
There may be a duplicate event sent, in my DB:
| 8723082022949765307 | 2023-07-06 15:42:19 | deleteSubject.event            | 2023-07-06 15:42:19 |                                                                                                                                    | 3                       |     NULL |      1 |     NULL |       1 |
| 8934052352645808926 | 2023-07-06 15:42:18 | deleteSubject.event            | 2023-07-06 15:42:18 |                                                                                                                                    | 3                       |     NULL |      1 |     NULL |       1 |

And I have an error in logs:
datasets                  | org.springframework.amqp.rabbit.support.ListenerExecutionFailedException: Listener method 'public void org.shanoir.ng.configuration.amqp.RabbitMQDatasetsService.deleteSubject(java.lang.String) throws org.springframework.amqp.AmqpRejectAndDontRequeueException' threw exception

Reply:
I've corrected this by using the service delete method, instead of the repository one.
Seems better now.